lib/ui/text.dart

```diff
@@ -1489,6 +1490,24 @@ class ParagraphBuilder extends NativeFieldWrapperClass2 {
   }
   String _addText(String text) native 'ParagraphBuilder_addText';
 
+  /// Adds a space for a widget to the paragraph.
+  ///
+  /// The paragraph will contain a rectangular space with no text of the size
+  /// specified.
+  ///
+  /// The [width] and [height] parameters specify the size of the rectangle, and
+  /// [baseline] indicates the distance of the alphabetic baseline from the top of
+  /// of the rectangle. This determines how the rectangle is aligned to the
+  /// surrounding text.
+  ///
+  /// For example, to make a 50x50 rectangle that sits completely on top of the
+  /// alphabetic baseline, pass in `addWidget(50, 50, 50)`. For a box that is
+  /// vertically centered on the alphabetic baseline, pass in `addWidget(50, 50, 25);`.
+  void addWidget(double width, double height, double baseline) {
```

Comment:
Docs should mention if there's a line breaking opportunity between two adjacent holes.

Reply:
Currently, we add break opportunities around each hole. I plan to add properties to support user-controlled breaking behavior, but would prefer to not do so in this initial version in order to avoid much larger necessary patches to minikin.

Reply:
i don't have a strong opinion on what the behaviour should be, just that it be documented. :-)